### PR TITLE
[BugFix] Fix HDFS stat information lost in profile and avoid BE crashed when get HDFS stat

### PR DESCRIPTION
**Pull request by Smith-Cruise**
Fix HDFS stat information lost in the profile and avoid BE crashing when get HDFS stat.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/io/shared_buffered_input_stream.cpp [hunk 1]
```diff
@@ -20,7 +20,7 @@ namespace starrocks::io {
 
 SharedBufferedInputStream::SharedBufferedInputStream(std::shared_ptr<SeekableInputStream> stream,
                                                      const std::string& filename, size_t file_size)
```
Comment (dirtysalt) [inline be/src/io/shared_buffered_input_stream.cpp:22]: if it's reference, then no need to use move.

Comment (dirtysalt) [reply be/src/io/shared_buffered_input_stream.cpp:22]: otherwise reference data will be clean.

File: be/src/io/shared_buffered_input_stream.h [hunk 2]
```diff
@@ -53,6 +53,10 @@ class SharedBufferedInputStream : public SeekableInputStream {
         return _stream->skip(count);
     }
 
+    StatusOr<std::unique_ptr<NumericStatistics>> get_numeric_statistics() override {
+        return _stream->get_numeric_statistics();
```
Comment (DorianZheng) [inline be/src/io/shared_buffered_input_stream.h:57]: Is this why we don't get hdfs profile now?

Comment (Smith-Cruise) [reply be/src/io/shared_buffered_input_stream.h:57]: yes, `SharedBufferedInputStream` didn't override `get_numeric_statistics()` function.

File: be/src/io/shared_buffered_input_stream.cpp (hunk 1 above)
Comment (Smith-Cruise) [reply be/src/io/shared_buffered_input_stream.cpp:22]: fixed

**Review by letian-jiang — APPROVED** [review]

File: be/src/io/shared_buffered_input_stream.cpp (hunk 1 above)
Comment (Smith-Cruise) [reply be/src/io/shared_buffered_input_stream.cpp:22]: actually `std::move()` can't change const reference data

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**